cmd/gpu_nfdhook/labeler.go

```diff
@@ -245,40 +282,120 @@ scanning:
 	}
 }
 
+// this returns pci groups label value, groups separated by "_", gpus separated by ".".
+// Example for two groups with 4 gpus: "0.1.2.3_4.5.6.7".
+func (l *labeler) createPCIGroupLabel(gpuNumList []string) string {
+	pciGroups := map[string][]string{}
+
+	pciGroupLevel := getEnvVarNumber(pciGroupingEnv)
+	if pciGroupLevel == 0 {
+		return ""
+	}
+
+	for _, gpuNum := range gpuNumList {
+		symLinkTarget, err := filepath.EvalSymlinks(path.Join(l.sysfsDRMDir, "card"+gpuNum))
+
+		if err == nil {
+			if pathPart := getPCIPathParts(pciGroupLevel, symLinkTarget); pathPart != "" {
+				pciGroups[pathPart] = append(pciGroups[pathPart], gpuNum)
+			}
+		}
```

Comment:
`NewPCIDevice` in `pkg/fpga` looks very similar (very quick look). could some of that be reused?